client/files.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// MakeDirOptions holds the options for a call to ListFiles.
+type MakeDirOptions struct {
+	// Path is the absolute path of the directory to be created (required).
+	Path string
+
+	// MakeParents specifies whether the non-existing parent directories will
+	// be created or not. If one of the parent directories of the specified
+	// path do not exist and MakeParents is set to false, the call will fail.
+	MakeParents bool
+
+	// Permissions specifies the 3-digit octal UNIX permissions the created
+	// directories will have.
+	Permissions string
+
+	// UserID indicates the ID of the owner user for the created directories.
+	UserID *int
+
+	// User indicates the name of the owner user for the created directories.
+	// If used together with UserID, UserID takes precedence.
```

Comment:
Per comment below, I think we should make this match the API, and behave like "If used together with UserID, the value must match UserID's username."